### PR TITLE
feat: introduces service virtualization using Hoverfly

### PR DESCRIPTION
**Pull request by bartoszmajsak**
This PR brings service virtualization based on [Hoverfly]() to set us free from 3rd party calls during tests (in this case GitHub).

The heart of it is [`HoverflyRuleConfigurer`](https://github.com/fabric8-launcher/launcher-backend/blob/31e6bcd2d450999bda2a903bee6d24271dc0df56/services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java) where the functionality is described in details. 

This setup requires one extra env variable `LAUNCHER_TESTS_TRUSTSTORE_PATH` which should be pointing to `hoverfly.jks` which comes as part of this PR. We will resolve it on `arquillian-core` side soon.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -270,6 +271,17 @@
         <artifactId>value</artifactId>
         <version>${version.immutables}</version>
       </dependency>
+      <dependency>
```
Comment (gastaldi) [inline pom.xml]: Why are you declaring the `booster-catalog-service` dependency again?

File: services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                .proxyPort(port);
+
+        if (Boolean.valueOf(getEnvVarOrSysProp("LAUNCHER_TESTS_SV_SIMULATION", "true"))) {
+            return HoverflyRule.inSimulationMode(defaultPath(simulationFile), hoverflyProxyConfig);
+        } else {
+            return HoverflyRule.inCaptureMode("captured-" + simulationFile, hoverflyProxyConfig);
+        }
+    }
+
+    private static String getEnvVarOrSysProp(String name, String defaultValue) {
```
Comment (gastaldi) [inline services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java]: Can we use `EnvironmentSupport.INSTANCE.getEnvVarOrSysProp` here?

File: pom.xml (hunk 1 above)
Comment (bartoszmajsak) [reply pom.xml]: Probably a merge conflict. Fixed

File: services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java (hunk 2 above)
Comment (bartoszmajsak) [reply services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java]: Done.

File: README.md [hunk 3]
```diff
@@ -106,7 +105,14 @@ Prerequisites to Run Integration Tests
         export LAUNCHER_MISSIONCONTROL_OPENSHIFT_PASSWORD=<pass>
       ```
 
-4. (Optional) Ensure from the previous steps all environment variables are properly set up and sourced into your terminal:
+4. Testing setup
+
+   * Make sure you refer to a trust store used for Service Virtualization tests:
+     ```
+     export LAUNCHER_TESTS_TRUSTSTORE_PATH=${PWD}/services/git-service-impl/src/test/resources/hoverfly/hoverfly.jks
```
Comment (gastaldi) [inline README.md]: There is a TLDR; version of this script a few lines below. Maybe it would be nice to have in that place too

Comment (bartoszmajsak) [reply README.md]: Done

File: services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java [hunk 4]
```diff
@@ -29,15 +30,11 @@ public static HoverflyRule createHoverflyProxy(String simulationFile, String des
                 .destination(destination)
                 .proxyPort(port);
 
-        if (Boolean.valueOf(getEnvVarOrSysProp("LAUNCHER_TESTS_SV_SIMULATION", "true"))) {
+        if (Boolean.valueOf(EnvironmentSupport.INSTANCE.getEnvVarOrSysProp("LAUNCHER_TESTS_SV_SIMULATION", "true"))) {
```
Comment (gastaldi) [inline services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java]: Ahem `EnvironmentSupport.INSTANCE.getBooleanEnvVarOrSysProp` :)

Comment (bartoszmajsak) [reply services/git-service-impl/src/test/java/io/fabric8/launcher/service/github/test/HoverflyRuleConfigurer.java]: Ack, fixed too.

**Comment by gastaldi** [comment]
Are the following properties still required to run the tests after this change? 

- `LAUNCHER_MISSIONCONTROL_GITHUB_USERNAME`
- `LAUNCHER_MISSIONCONTROL_GITHUB_TOKEN`
- `LAUNCHER_MISSIONCONTROL_GITLAB_USERNAME`
- `LAUNCHER_MISSIONCONTROL_GITLAB_PRIVATE_TOKEN`

**Comment by gastaldi** [comment]
I see a new file is created after the tests are run (probably should be added to .gitignore?): `services/git-service-impl/src/test/resources/hoverfly/captured-gh-simulation.json`

**Comment by gastaldi** [comment]
Thank you!

**Pull request merged**